static/src/javascripts/projects/commercial/modules/third-party-tags.ts

```diff
@@ -51,23 +54,29 @@ const addScripts = (tags) => {
 		tag.loaded = true;
 	});
 
+	// eslint-disable-next-line @typescript-eslint/no-unnecessary-condition -- false positive
 	if (hasScriptsToInsert) {
-		fastdom.mutate(() => {
-			if (ref && ref.parentNode) {
+		void fastdom.mutate(() => {
+			if (ref.parentNode) {
 				ref.parentNode.insertBefore(frag, ref);
 			}
 		});
 	}
 };
 
 const insertScripts = (
-	advertisingServices,
-	performanceServices, // performanceServices always run
-) => {
+	advertisingServices: ThirdPartyTag[],
+	performanceServices: ThirdPartyTag[], // performanceServices always run
+): Promise<void> => {
 	addScripts(performanceServices);
-	void getInitialConsentState().then((state) => {
+	return getInitialConsentState().then((state) => {
 		const consentedAdvertisingServices = advertisingServices.filter(
-			(script) => getConsentFor(script.name, state),
+			(script) => {
+				// TODO - is this the correct behavior?
+				// throw error if script name undefined?
+				if (script.name === undefined) return false;
+				return getConsentFor(script.name, state);
```

Comment:
or maybe just log a warning and return false?

Reply:
You mentioned below that this is probably because we don’t have the latest CMP version. I would be happy to keep this as an assertion for the time being: `(script.name as string)`

Reply:
As in `return getConsentFor(script.name as string, state);`? That seems to give me 
```
Argument of type 'string' is not assignable to parameter of type '"a9" | "acast" | "braze" | "comscore" | "facebook-mobile" | "fb" | "firebase" | "google-analytics" | "google-mobile-ads" | "google-sign-in" | "google-tag-manager" | "googletag" | ... 13 more ... | "youtube-player"'.
```

Maybe its best just preserve the existing behavior for now, returning false if `script.name` is undefined.